### PR TITLE
Add Cloud IDE note

### LINKED ISSUE #3602: Add info about public accessibility for Cloud IDE integrations
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

- [ ] https://docs.astronomer.io/astro/cloud-ide/quickstart#step-2-create-a-pipeline
- [x] https://docs.astronomer.io/astro/cloud-ide/configure-project-environment

### The Issue or Context

In order to connect a pipeline with GitLab, the GitLab or other 3rd party resource must be public in order to connect with the Cloud IDE. 

### Fix or Proposed Change

Add note specifying limitation.

### Additional Notes

https://astronomer.slack.com/archives/C01UA03TPV1/p1712167303027819

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #3602

### CODE REVIEW

File: astro/cloud-ide/configure-project-environment.md [hunk 1]
```diff
@@ -7,13 +7,14 @@ description: Learn how to configure environment variables, connections, and depe
...
 
-Environment configuration in the Astro Cloud IDE is similar to Astro project and Airflow configuration, but more options are provided to minimize the time you spend configuring text files. 
+Environment configuration in the Astro Cloud IDE is similar to Astro project and Airflow configuration, but more options are provided to minimize the time you spend configuring text files.
 
 ## Limitations
 
 - You can't set OS-level dependencies with the Astro Cloud IDE.
 - Only a subset of Airflow connections can be configured through Astro Cloud IDE templates. All other connections must be configured through a generic connection template.
 - Environment configurations are not saved when you export a data pipeline as a DAG. To run your data pipeline outside of the Astro Cloud IDE, you must reconfigure these values in the Airflow environment where you exported your data pipeline.
+- In order to connect the Cloud IDE to a 3rd party resource, such as GitLab, the resource cannot be private. It must be public.
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:17]:
```suggestion
- To connect the Cloud IDE to a 3rd party resource, such as GitLab, the resource must be publicly accessible.
```

File: astro/cloud-ide/quickstart.md [hunk 2]
```diff
@@ -53,6 +53,12 @@ When you first run your project, your pipeline is built into a single DAG with t
 
 After clicking **Create**, the IDE opens the pipeline editor. This is where you'll write your pipeline code.
 
+:::note
```
Comment (jwitz) [inline astro/cloud-ide/quickstart.md:56]:
```suggestion
:::info
```

File: astro/cloud-ide/quickstart.md [hunk 3]
```diff
@@ -53,6 +53,12 @@ When you first run your project, your pipeline is built into a single DAG with t
...
+
+In order to connect the Cloud IDE to a 3rd party resource, such as GitLab, the resource cannot be private. It must be public.
```
Comment (jwitz) [inline astro/cloud-ide/quickstart.md:58]: Not sure we need the note in the quickstart? I don't see an opportunity for anyone to connect to a 3rd party resource in this setup.

**Review by jwitz — CHANGES_REQUESTED** [review]
Looking at the original Slack comment, I think this is more related to repository setup rather than connecting a data pipeline to an external resource. We should put this note in https://docs.astronomer.io/astro/cloud-ide/deploy-project#commit-your-project-to-a-git-repository instead

**Comment by lzdanski** [comment]
I think i've got the "format document" setting enabled on my VScode to address extra whitespace. So there's one file that shouldn't have any content changes, just removing trailing whitespace.

File: astro/cloud-ide/configure-project-environment.md [hunk 4]
```diff
@@ -7,13 +7,14 @@ description: Learn how to configure environment variables, connections, and depe
...
 
-Environment configuration in the Astro Cloud IDE is similar to Astro project and Airflow configuration, but more options are provided to minimize the time you spend configuring text files. 
+Environment configuration in the Astro Cloud IDE is similar to Astro project and Airflow configuration, but more options are provided to minimize the time you spend configuring text files.
 
 ## Limitations
 
 - You can't set OS-level dependencies with the Astro Cloud IDE.
 - Only a subset of Airflow connections can be configured through Astro Cloud IDE templates. All other connections must be configured through a generic connection template.
 - Environment configurations are not saved when you export a data pipeline as a DAG. To run your data pipeline outside of the Astro Cloud IDE, you must reconfigure these values in the Airflow environment where you exported your data pipeline.
+- To connect the Cloud IDE to a 3rd party resource, such as GitLab, the resource must be publicly accessible.
```
Comment (jwitz) [inline astro/cloud-ide/configure-project-environment.md:17]:
```suggestion
```

File: astro/cloud-ide/deploy-project.md [hunk 5]
```diff
@@ -22,21 +22,27 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
+In order to connect the Cloud IDE to a 3rd party resource, such as GitLab, the resource cannot be private. It must be public.
+
+:::
+
+### Prerequisites
 
 - A Cloud IDE project. See the [Quickstart](/cloud-ide/quickstart.md#step-1-log-in-and-create-a-project).
-- A Git repository hosted with a supported Git vendor. 
-- A user account on your Git vendor with a personal access token (PAT). 
+- A Git repository hosted with a supported Git vendor.
```
Comment (jwitz) [inline astro/cloud-ide/deploy-project.md:42]:
```suggestion
- A Git repository hosted with a supported Git vendor. The Git repository must be publicly accessible.
```

File: astro/cloud-ide/deploy-project.md [hunk 6]
```diff
@@ -22,21 +22,27 @@ Astro Cloud IDE projects use the same structure as Astro projects created with t
...
 ## Commit your project to a Git repository
 
 The Cloud IDE includes Git vendor integrations for managing different versions and features of your Cloud IDE projects. Configuring a Git repository is also the first step to deploying your Cloud IDE project to Astro. Astronomer recommends setting up a separate Git repository for every Cloud IDE project.
 
-### Prerequisites 
+:::note
+
+In order to connect the Cloud IDE to a 3rd party resource, such as GitLab, the resource cannot be private. It must be public.
+
+:::
```
Comment (jwitz) [inline astro/cloud-ide/deploy-project.md:37]:
```suggestion
```
I think it makes more sense to keep this info close to the other prerequisites

**Review by jwitz — APPROVED** [review]

**Pull request merged**